itest/lnd_route_blinding_test.go

```diff
@@ -1510,3 +1510,83 @@ func testBlindedPaymentHTLCReForward(ht *lntest.HarnessTest) {
 		require.Fail(ht, "timeout waiting for sending payment")
 	}
 }
+
+// testGivenBlindedRouteInvoices tests lnd's ability to create a blinded
+// payment path when the income node is given. Which it then inserted into
+// an invoice, sent to an invoice with a blinded path and forwarded payment
+// in a blinded route and finally, receiving the payment.
+func testGivenBlindedRouteInvoices(ht *lntest.HarnessTest) {
+	var (
+		// Convenience aliases.
+		alice = ht.Alice
+		bob   = ht.Bob
+	)
+
+	// Setup a two hop channel network: Alice -- Bob -- Carol.
+	// We set our proportional fee for these channels to zero, so that
+	// our calculations are easier. This is okay, because we're not testing
+	// the basic mechanics of pathfinding in this test.
+	chanAmt := btcutil.Amount(100000)
+	chanPointAliceBob := ht.OpenChannel(
+		alice, bob, lntest.OpenChannelParams{
+			Amt:        chanAmt,
+			BaseFee:    10000,
+			FeeRate:    0,
+			UseBaseFee: true,
+			UseFeeRate: true,
+		},
+	)
+
+	carol := ht.NewNode("Carol", nil)
+	ht.EnsureConnected(bob, carol)
+
+	var bobCarolBase uint64 = 2000
+	chanPointBobCarol := ht.OpenChannel(
+		bob, carol, lntest.OpenChannelParams{
+			Amt:        chanAmt,
+			BaseFee:    bobCarolBase,
+			FeeRate:    0,
+			UseBaseFee: true,
+			UseFeeRate: true,
+		},
+	)
+
+	// Wait for Alice to see Bob/Carol's channel because she'll need it for
+	// pathfinding.
+	ht.AssertTopologyChannelOpen(alice, chanPointBobCarol)
+
+	// Lookup full channel info so that we have channel ids for our route.
+	bobCarolChan := ht.GetChannelByChanPoint(bob, chanPointBobCarol)
+
+	// Let Carol add a blinded invoice choosing the income channel.
+	var (
+		minNumRealHops      uint32 = 1
```

Comment:
will this be reseted if set to 2 for example but the chainedChannel is of length 1 ?

Reply:
maybe we can add either a itest case or a unit test to test this behviour.

Reply:
Yes, for sure, but we would expanding the [specification](https://github.com/lightningnetwork/lnd/pull/9127#issuecomment-2435692652) agreed before (item 6).

Reply:
ok so let's leave it for now then and we decide later (maybe you can add this test into the unit test part so we don't need the itest for this case)